### DISCUSSION TITLE
Swagger and OpenAPI plugins

### DISCUSSION THREAD

**Discussion by teunmooij** [payloadcms/payload#2432] [Show and tell]
Hi, we've just released the first major version of our [payload-swagger](https://www.npmjs.com/package/payload-swagger) and [payload-openapi](https://www.npmjs.com/package/payload-openapi) plugins.

With `payload-openapi` you can generate openAPI 3.0 documentation for your payload instance, which includes:
- collection endpoints
- global endpoints
- custom endpoints
- authentication endpoints
- preferences endpoints
- fully typed schema definitions for all requests, parameters and responses
- authentication requirements for all your endpoints
- extension points to merge your custom openapi definitions into the schema

The `payload-swagger` plugin turns `payload-openapi` into a simple plugin that not only generates the openAPI document, but also adds a swagger UI endpoint to your payload server.

You can find them here:
- `payload-swagger`: https://www.npmjs.com/package/payload-swagger
- `payload-openapi`: https://www.npmjs.com/package/payload-openapi
- sources: https://github.com/teunmooij/payload-tools

**Comment by RafaelPaulo12** [comment]
Hi.
Is there any prediction for an update of the payload-swagger and payload-openapu plugins for Payload version 3 ?

**Reply by janbuchar** [reply]
Hi, I'm working on payload 3.0 support for my openapi plugin - https://github.com/janbuchar/payload-oapi/pull/18. Feel free to check it out :slightly_smiling_face:

**Reply by jeanlescure** [reply]
@janbuchar you are the real MVP

I had recently mentioned this as one of my team's pain points with PayloadCMS: 

https://github.com/payloadcms/payload/discussions/11213#discussioncomment-12419584

I just tried your plugin, it almost works perfectly, in our case even when developing locally we use a dummy domain with https which caused a CORS/MIME issues with `unpkg`, I opened a Pull Request with the fix:

https://github.com/janbuchar/payload-oapi/pull/24

Thank you and keep up the awesome work! 🚀

**Reply by janbuchar** [reply]
Hi @jeanlescure ! It's great to hear that you find that plugin useful, I'm super happy for that 🙂

**Reply by vipuljbhikadiya** [reply]
@janbuchar plugins is awesome. does this support auth paths in any collection? or is that in roadmap ?

**Reply by janbuchar** [reply]
@vipuljbhikadiya I'm not sure I follow, can you show me an example config? Or feel free to open an issue in the payload-oapi repo.
